### PR TITLE
Mention Intersection Observer v2 in sec considerations

### LINKED ISSUE #1105: Refer to IntersectionObserver from the Security Considerations
IntersectionObserver1 and/or [IntersectionObserver2](https://szager-chromium.github.io/io-v2/) could be valuable to relying parties to determine whether an iframe they are utilizing for WebAuthn is being clearly shown to the user as intended. We should consider adding text to the Security Considerations section referring to the appropriate (final) version(s) of this.

### PR DESCRIPTION
**Pull request by equalsJeffH**
fixes #1105 by describing threat of UI redressing aka clickjacking and pointing to (experimental) https://w3c.github.io/IntersectionObserver/v2/.

note: this PR moves two existing sec cons sections around and inserts a new section -- nothing already existing is removed.

***

### :boom: Error: 500 Internal Server Error :boom: ###

[PR Preview](https://github.com/tobie/pr-preview#pr-preview) failed to build. _(Last tried on Oct 21, 2020, 8:49 PM UTC)_.

<details>
<summary>More</summary>

PR Preview relies on a number of web services to run. There seems to be an issue with the following one:

:rotating_light: [CSS Spec Preprocessor](https://api.csswg.org/bikeshed/) - CSS Spec Preprocessor is the web service used to build Bikeshed specs.

:link: [Related URL](https://api.csswg.org/bikeshed/?url=https%3A%2F%2Fraw.githubusercontent.com%2Fw3c%2Fwebauthn%2Fe5dd87b55f5c83588f7534192b5e94a073105950%2Findex.bs&force=1&md-warning=not%20ready)

```
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>500 Internal Server Error</title>
</head><body>
<h1>Internal Server Error</h1>
<p>The server encountered an internal error or
misconfiguration and was unable to complete
your request.</p>
<p>Please contact the server administrator at 
 [no address given] to inform them of the time this error occurred,
 and the actions you performed just before this error.</p>
<p>More information about this error may be available
in the server error log.</p>
<hr>
<address>Apache/2.4.10 (Debian) Server at api.csswg.org Port 443</address>
</body></html>

```

_If you don't have enough information above to solve the error by yourself (or to understand to which web service the error is related to, if any), please [file an issue](https://github.com/tobie/pr-preview/issues/new?title=Error%20not%20surfaced%20properly&body=See%20w3c/webauthn%231497.)._
</details>

### CODE REVIEW

**Review by agl — APPROVED** [review]

File: index.bs [hunk 1]
```diff
@@ -6529,8 +6517,30 @@ The main benefits offered to [=[WRPS]=] by this specification include:
...
+
+Simplistic use of WebAuthn in an embedded context, e.g., within <{iframe}>s as described in [[#sctn-iframe-guidance]], may make users vulnerable to <dfn>UI Redressing</dfn> attacks, also known as "[Clickjacking](https://en.wikipedia.org/wiki/Clickjacking)". This is where an attacker overlays their own UI on top of a [=[RP]=]'s intended UI and attempts to trick the user into performing unintended actions with the [=[RP]=]. For example, using these techniques, an attacker might be able to trick users into purchasing items, transferring money, etc.
+
+Therefore, it is important for an [=[RP]=] having embedded WebAuthn-wielding content to ensure that their content's UI is visible to the user. An emerging means to do so is by observing the status of the experimental [Intersection Observer v2](https://w3c.github.io/IntersectionObserver/v2/)'s `isVisible` attribute. For example, the [=[RPS]=] script running in the embedded context could pre-emptively load itself in a popup window if it detects `isVisble` being set to `false`, thus side-stepping any occluding of their content that may be occuring over the embedded content.
```
Comment (emlun) [inline index.bs:6527]: I think the `[=[RPS]=]` alias is for the plural, right?

```suggestion
Therefore, it is important for an [=[RP]=] having embedded WebAuthn-wielding content to ensure that their content's UI is visible to the user. An emerging means to do so is by observing the status of the experimental [Intersection Observer v2](https://w3c.github.io/IntersectionObserver/v2/)'s `isVisible` attribute. For example, the [=[RP]=]'s script running in the embedded context could pre-emptively load itself in a popup window if it detects `isVisble` being set to `false`, thus side-stepping any occluding of their content that may be occuring over the embedded content.
```

**Review by emlun — CHANGES_REQUESTED** [review]

File: index.bs (hunk 1 above)
Comment (equalsJeffH) [reply index.bs:6527]: done, thx :)

**Review by akshayku — APPROVED** [review]

File: index.bs [hunk 2]
```diff
@@ -6529,8 +6517,30 @@ The main benefits offered to [=[WRPS]=] by this specification include:
...
+
+Even though WebAuthn-specific UI is typically handled by the [=client platform=] and thus is not vulnerable to [=UI Redressing=], it is likely important for an [=[RP]=] having embedded WebAuthn-wielding content to ensure that their content's UI is visible to the user. An emerging means to do so is by observing the status of the experimental [Intersection Observer v2](https://w3c.github.io/IntersectionObserver/v2/)'s `isVisible` attribute. For example, the [=[RP]=]'s script running in the embedded context could pre-emptively load itself in a popup window if it detects `isVisble` being set to `false`, thus side-stepping any occlusion of their content.
```
Comment (jcjones) [inline index.bs:6527]: nit: I love the description of "WebAuthn-wielding", but it's probably overly colorful for the document. Perhaps consider being plainer with just "WebAuthn-using"?

**Review by jcjones — APPROVED** [review]
Sorry to argue red v. blue, but approved from my end, just wanted to give the nit.

**Review by emlun — APPROVED** [review]

**Pull request merged**